planner/select.go

```diff
@@ -285,7 +286,29 @@ func (n *selectNode) initSource() ([]aggregateNode, error) {
 		}
 	}
 
-	return n.initFields(n.selectReq)
+	aggregates, err := n.initFields(n.selectReq)
+	if err != nil {
+		return nil, err
+	}
+
+	if isScanNode {
+		origScan.initFetcher(n.selectReq.Cid, findFilteredByIndexedField(origScan))
+	}
+
+	return aggregates, nil
+}
+
+func findFilteredByIndexedField(scanNode *scanNode) immutable.Option[client.FieldDescription] {
+	if scanNode.filter != nil {
+		indexedFields := scanNode.desc.CollectIndexedFields()
+		for i := range indexedFields {
+			typeIndex := scanNode.documentMapping.FirstIndexOfName(indexedFields[i].Name)
+			if scanNode.filter.HasIndex(typeIndex) {
+				return immutable.Some(indexedFields[i])
```

Comment:
because the first implementation should be relatively simple.
More sophisticated analysis and optimisations will come later.
There is a bunch of things that can be optimised here, but implementing them all at once would make this PR humongous.

Reply:
Ah, this is us deliberately ignoring other indexes for now?

Can you please document that in the code?